docs/API/v3/package-publish-resource.md

```diff
@@ -0,0 +1,112 @@
+---
+# required metadata 
+
+title: Push and Delete | NuGet V3 API | Microsoft Docs
+author:
+- joelverhagen
+- kraigb
+ms.author:
+- joelverhagen
+- kraigb
+manager: skofman
+ms.date: 10/26/2017
+ms.topic: reference
+ms.prod: nuget
+ms.technology: null
+ms.assetid: 1eaa403a-5c13-4c05-9352-2f791b98aa7e
+
+# optional metadata
+
+description: The publish service allows clients to publish new packages and unlist or delete existing packages.
+keywords: NuGet, API, REST, HTTP, push, delete, unlist, package, upload, create, hide
+ms.reviewer:
+- karann
+- unniravindranathan
+
+---
+
+# Push and Delete
+
+It is possible to push and delete (or unlist, depending on the server implementation) packages using the NuGet V3 API.
+Both operations are based off of the `PackagePublish` resource found in the [service index](service-index.md).
+
+## Versioning
+
+The only `@type` of the resource used for pushing and deleting packages is: `PackagePublish/2.0.0`.
+
+## Base URL
+
+The base URL for the following APIs is the value of the `@id` property of the `PackagePublish/2.0.0` resource in the
+package source's [service index](service-index.md). For the documentation below, NuGet.org's URL is used. Consider 
+`https://www.nuget.org/api/v2/package` as a placeholder for the `@id` value found in the service index.
+
+Note that this URL points to the same location as the legacy V2 push endpoint since the protocol for V2 and V3 push is
+the same.
+
+## HTTP methods
+
+The `PUT` and `DELETE` HTTP methods are supported by this resource. For which methods are supported on each endpoint,
+see below.
+
+## Push a package
+
+NuGet.org supports pushing new packages using the following API. If the package with the provided ID and version
+already exists, NuGet.org will reject the push. Other package sources may support replacing an existing package.
+
+```
+PUT https://www.nuget.org/api/v2/package
+```
+
+### Request parameters
+
+Name           | In     | Type   | Notes
+-------------- | ------ | ------ | -----
+X-NuGet-ApiKey | Header | string | Required: for example, `X-NuGet-ApiKey: {USER_API_KEY}`
+
+### Request body
+
+The request body can come in one of two forms.
+
+#### Multipart form data
+
+The request header `Content-Type` is `multipart/form-data` and the first item in the response body is the raw bytes of
+the .nupkg being pushed. Subsequent items in the multipart body are ignored.
+
+#### Raw bytes
+
+If the response body is not `multipart/form-data`, the entire request body is assumed to be the raw bytes of the
+.nupkg being pushed.
+
+### Response
+
+Status Code | Meaning
+----------- | -------
+200, 201    | The package was successfully pushed
+400         | The provided package is invalid
+409         | A package with the provided ID and version already exists
+
+## Delete a package
+
+NuGet.org interprets the package delete request as an "unlist". This means that the package is still available for
```

Comment:
What is the policy for other implementors?
At MyGet we took unlist as well to have consistency with NuGet, but not sure the sam semantics should be followed everywhere?

Reply:
NuGet.Server is capable of handling `DELETE` as a hard delete. I'll explain more here about the general policy.

Reply:
Fixed.